albogdan/skule-vote
# Create modal to confirm that you haven't ranked all candidates

## Overview

- If you click "Cast Ballot" without ranking every candidate, you will see the following: <img width="413" alt="Screen Shot 2022-02-26 at 1 56 51 PM" src="https://user-images.githubusercontent.com/42653157/155855707-e71c36cb-2dea-4f87-b83d-5252a841eebe.png">
- Also switched a lot if styles from "styled-components" to MUI's built in `styled()`
- Relocated components and styled from "BallotModal.js" cause that file got WAY too big
- Changed multi seat prompt from "Please select as many choices as you want using the dropdown menus." to "Please select as many choices as you want using the dropdown menus. You are encouraged to rank all of the above choices"


## Unit Tests Created

- Create unit test for new modal
- Create new unit tests in BallotModal.test.js to confirm the modal comes out when expected


## Steps to QA

- In a single candidate election / referendum:
   - Modal should never appear
- In a multi candidate eleection:
   - If you rank every candidate, clicking "Cast ballot" button should just submit your vote
   - If you don't rank all candidates, clicking "Cast ballot" button should make the modal appear. Make sure the modal's buttons close the modal or submit the modal
- Please check on mobile and dark mode for style issues
